_source/_posts/2020-08-07-spring-boot-remote-vs-local-tokens.md

```diff
@@ -0,0 +1,408 @@
+---
+layout: blog_post
+title: "JWT vs Opaque Access Tokens: Use Both With Spring Boot"
+author: brian-demers
+by: advocate
+communities: [java]
+description: "Tutorial: Learn how to use JWT and opaque access with Spring Boot."
+tags: [java, spring, spring-boot, security, oauth, jwt]
+tweets:
+- "JWTs are fast and stale, Opaque tokens are slow and secure. Use both for the best of both worlds. #SpringBoot 🌎"
+- "Learn how to use JWT and opaque access tokens in the same app with @SpringBoot 💥"
+- "Use JWTs for reads (fast) and opaque (secure) tokens for writes with your #SpringBoot application 🔒"
+image: blog/spring-boot-remote-vs-local-tokens/spring-jwt-and-opaque.png
+type: conversion
+---
+The topic of validating an OAuth 2.0 access tokens comes up frequently on the Okta developer blog. Often we talk about how to validate JSON Web Token (JWT) based access tokens; however, this is NOT part of the OAuth 2.0 specification. JWTs are so commonly used that Spring Security supported them before adding support for remotely validating tokens (which is part of the OAuth 2.0 specification.)
```

Comment:
"the Okta developer blog" seems funny. I'd change it to "this blog" or "our developer forums".

Reply:
OH good call, I had to change that when it was on Scotch